File: PHP/70.爬楼梯.php
/*
 * @lc app=leetcode.cn id=70 lang=php
 *
 * [70] 爬楼梯
 */

// @lc code=start
class Solution {
    public $num;
    /**
     * @param Integer $n
     * @return Integer
     */
    function climbStairs($n) {
        if($n < 3){
            return $n;
        }
        $num[1] = 1;
        $num[2] = 2;
        for($i = 3;$i<=$n;$i++){
            $num[$i] = $num[$i-1] + $num[$i-2];
        }
        return $num[$n];
    }
}
// @lc code=end

File: JAVA/70.爬楼梯.java
/*
 * @lc app=leetcode.cn id=70 lang=java
 *
 * [70] 爬楼梯
 */

// @lc code=start
class Solution {
    public int climbStairs(int n) {
        if(n==1){
            return 1;
        }
        if(n==2){
            return 2;
        }
        int[] result = new int[n + 1];
        result[0]=0;
        result[1]=1;
        result[2]=2;
        for(int i=3;i<=n;i++){
            result[i] = result[i-1]+result[i-2];
        }
        return result[n];
    }
}
// @lc code=end

